fdbserver/workloads/MachineAttrition.actor.cpp

```diff
@@ -173,51 +188,39 @@ struct MachineAttritionWorkload : TestWorkload {
-				}
-			}
+			TraceEvent("Assassination").detail("TargetDataCenterIds", describe(self->targetIds));
+			sendRebootRequests(workers, self->targetIds, rbReq,
+			                   // idAccess lambda
+			                   [](WorkerDetails worker) { return worker.interf.locality.dcId(); });
 		} else if (self->killMachine) {
-			wait(delay(delayBeforeKill));
-			// Pick a machine to kill
-			Optional<Standalone<StringRef>> killMachineId = self->targetId.toString().empty() ? workers.back().interf.locality.machineId() : self->targetId;
-			TraceEvent("Assassination").detail("TargetMachineId", killMachineId);
-			for (const auto& worker : workers) {
-				// kill all matching machine workers
-				if (worker.interf.locality.machineId().present() && worker.interf.locality.machineId() == killMachineId) {
-					TraceEvent("SendingRebootRequest").detail("TargetMachine", worker.interf.locality.toString());
-					worker.interf.clientInterface.reboot.send(rbReq);
-				}
-			}
+			TraceEvent("Assassination").detail("TargetMachineIds", describe(self->targetIds));
+			sendRebootRequests(workers, self->targetIds, rbReq,
+			                   // idAccess lambda
+			                   [](WorkerDetails worker) { return worker.interf.locality.machineId(); });
 		} else if (self->killDatahall) {
-			wait(delay(delayBeforeKill));
-			// Pick a datahall to kill
-			Optional<Standalone<StringRef>> killDatahallId = self->targetId.toString().empty() ? workers.back().interf.locality.dataHallId() : self->targetId;
-			TraceEvent("Assassination").detail("TargetDatahallId", killDatahallId);
-			for (const auto& worker : workers) {
-				// kill all matching datahall workers
-				if (worker.interf.locality.dataHallId().present() && worker.interf.locality.dataHallId() == killDatahallId) {
-					TraceEvent("SendingRebootRequest").detail("TargetMachine", worker.interf.locality.toString());
-					worker.interf.clientInterface.reboot.send(rbReq);
-				}
-			}
+			TraceEvent("Assassination").detail("TargetDatahallIds", describe(self->targetIds));
+			sendRebootRequests(workers, self->targetIds, rbReq,
+			                   // idAccess lambda
+			                   [](WorkerDetails worker) { return worker.interf.locality.dataHallId(); });
+		} else if (self->killProcess) {
+			TraceEvent("Assassination").detail("TargetProcessIds", describe(self->targetIds));
+			sendRebootRequests(workers, self->targetIds, rbReq,
+			                   // idAccess lambda
+			                   [](WorkerDetails worker) { return worker.interf.locality.processId(); });
+		} else if (self->killZone) {
+			TraceEvent("Assassination").detail("TargetZoneIds", describe(self->targetIds));
+			sendRebootRequests(workers, self->targetIds, rbReq,
+			                   // idAccess lambda
+			                   [](WorkerDetails worker) { return worker.interf.locality.zoneId(); });
 		} else {
 			while (killedMachines < self->machinesToKill && workers.size() > self->machinesToLeave) {
```

Comment:
This else block is used to randomly kill workers instead of machines. For example, killedMachines is actually used as killedWorkers. So it would be better to look at this else block and use variable names referring to workers instead of machines.

Reply:
I made the change, although the self->machine... parameters had to stay because they are used in the simulated versions of machineAttrition. Is that fine or is it preferable to have new parameters here too?

Reply:
It's better to have separate parameters. Actually looking at how this has evolved, there is not a lot of common code between MachineAttrition in simulation and read world killing. Probably it would be cleaner to have separate workloads. But i think it would be fine to just have separate parameters for now.